integration/assist/command_test.go

```diff
@@ -0,0 +1,458 @@
+	teleportDataDir := filepath.Join(testDir, "teleport_openssh")
+
+	openSSHCfg := servicecfg.MakeDefaultConfig()
+
+	openSSHCfg.OpenSSH.Enabled = true
+	err := config.ConfigureOpenSSH(&config.CommandLineFlags{
+		DataDir:           teleportDataDir,
+		ProxyServer:       rc.Web,
+		AuthToken:         testToken,
+		JoinMethod:        string(types.JoinMethodToken),
+		OpenSSHConfigPath: opensshConfigPath,
+		RestartOpenSSH:    false,
+		CheckCommand:      "echo okay",
+		Labels:            "hello=true",
+		Address:           sshAddr,
+		InsecureMode:      true,
+		Debug:             true,
+	}, openSSHCfg)
+	require.NoError(t, err)
+
+	err = service.Run(ctx, *openSSHCfg, nil)
+	require.NoError(t, err)
+
+	// Wait for node propagation
+	require.Eventually(t, helpers.FindNodeWithLabel(t, ctx, rc.Process.GetAuthServer(), "hello", "true"), time.Second*2, time.Millisecond*50)
+	nodes, err := rc.Process.GetAuthServer().GetNodes(ctx, apidefaults.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(nodes))
+	return nodes[0]
+}
+
+func setupMockSSHNode(t *testing.T, ctx context.Context, sshListener net.Listener, nodeName string, rc *helpers.TeleInstance) {
+	// Setup: creating and starting openssh mock server
+	ca, err := rc.Process.GetAuthServer().GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.HostCA,
+		DomainName: testClusterName,
+	}, true)
+	require.NoError(t, err)
+
+	signers, err := sshutils.GetSigners(ca)
+	require.NoError(t, err)
+	require.Len(t, signers, 1)
+
+	cert, err := apisshutils.MakeRealHostCertWithPrincipals(signers[0], nodeName)
+	require.NoError(t, err)
+	handler := sshutils.NewChanHandlerFunc(handlerSSH)
+	sshServer, err := sshutils.NewServer(
+		"test",
+		utils.NetAddr{AddrNetwork: "tcp", Addr: sshListener.Addr().String()},
+		handler,
+		[]ssh.Signer{cert},
+		sshutils.AuthMethods{
+			NoClient: true,
+		},
+		sshutils.SetInsecureSkipHostValidation(),
+		sshutils.SetLogger(utils.NewLoggerForTests().WithField("component", "mocksshserver")),
+	)
+	require.NoError(t, err)
+	require.NoError(t, sshServer.SetListener(sshListener))
+	require.NoError(t, sshServer.Start())
+}
+
+// this is a dummy SSH handler. It only supports "exec" requests. All other
+// requests are happily acknowledged and discarded. Receieving an "exec" request
+// sends testCommandOutput in the main channel and closes all channels.
+// This is not strictly following the SSH RFC as request processing is blocked
+// as soon as an exec request is received, but is good enough for our use-case.
+func handlerSSH(ctx context.Context, ccx *sshutils.ConnectionContext, nch ssh.NewChannel) {
+	ch, requests, err := nch.Accept()
+	if err != nil {
+		return
+	}
+	// Sessions have out-of-band requests such as "shell",
+	// "pty-req", "env" and "exec". Here we don't output anything and start a
+	// routine consuming requests and waiting for the "exec" one.
+	go func(in <-chan *ssh.Request) {
+		for req := range in {
```

Comment:
What's going to close the `requests` channel so this goroutine eventually terminates?